### ISSUE TITLE
newproxy() without an argument reports a missing parameter

### ISSUE THREAD

**Issue by ligurio** [EmmyLuaLs/emmylua-analyzer-rust#1249]
Imagine a file `newproxy.lua` with the following Lua chunk:

```lua
local p = newproxy()
print(p)
```

and the following config `.emmyrc.json` for the emmylua_check: `{ "runtime": { "version": "LuaJIT" } }`.

Run the command: `emmylua_check --config .emmyrc.json newproxy.lua`
The command reports:

```
---  [1 warning]
warning: expected 1 parameters but found 0. missing parameter: proxy [missing-parameter]
  -->: :1:20

  1 │ local p = newproxy()
  2 │ print(p)



Summary
  1 warning

Check completed with warnings
Check finished

```

It is expected no diagnostics.

It seems in LuaJIT `newproxy([proxy])` the argument is optional; `newproxy()` creates a userdata without a metatable.

Version: emmylua_check 0.24.0

LuaJIT behaviour:

```
$ luajit proxy.lua 
userdata: 0x7b77a67c90a8
$ luajit -v
LuaJIT 2.1.1703358377 -- Copyright (C) 2005-2023 Mike Pall. https://luajit.org/
```
